packages/mui-material/src/useMediaQuery/useMediaQuery.ts

```diff
@@ -42,11 +41,8 @@ function useMediaQueryOld(
   ssrMatchMedia: ((query: string) => { matches: boolean }) | null,
   noSsr: boolean | undefined,
 ): boolean {
-  const supportMatchMedia =
-    typeof window !== 'undefined' && typeof window.matchMedia !== 'undefined';
```

Comment:
Duplicated logic, we don't need that. Instead, we can directly test that `matchMedia` isn't nullish.